pkg/cmd/admin/migrate/images/images.go

```diff
@@ -0,0 +1,487 @@
+				continue
+			}
+			info.spec = updateString(&ref.From.Name, fn) || info.spec
+		}
+		for _, events := range t.Status.Tags {
+			for i := range events.Items {
+				info.status = updateString(&events.Items[i].DockerImageReference, fn) || info.status
+			}
+		}
+		return info, nil
+	case *kapi.Secret:
+		switch t.Type {
+		case kapi.SecretTypeDockercfg:
+			var v credentialprovider.DockerConfig
+			if err := json.Unmarshal(t.Data[kapi.DockerConfigKey], &v); err != nil {
+				return nil, err
+			}
+			if !updateDockerConfig(v, o.Mappings.MapDockerAuthKey) {
+				return reporter(false), nil
+			}
+			data, err := json.Marshal(v)
+			if err != nil {
+				return nil, err
+			}
+			t.Data[kapi.DockerConfigKey] = data
+			return reporter(true), nil
+		case kapi.SecretTypeDockerConfigJson:
+			var v credentialprovider.DockerConfigJson
+			if err := json.Unmarshal(t.Data[kapi.DockerConfigJsonKey], &v); err != nil {
+				return nil, err
+			}
+			if !updateDockerConfig(v.Auths, o.Mappings.MapDockerAuthKey) {
+				return reporter(false), nil
+			}
+			data, err := json.Marshal(v)
+			if err != nil {
+				return nil, err
+			}
+			t.Data[kapi.DockerConfigJsonKey] = data
+			return reporter(true), nil
+		default:
+			return reporter(false), nil
+		}
+	case *deployapi.DeploymentConfig:
+		changed := updatePodTemplate(t.Spec.Template, fn)
+		return reporter(changed), nil
+	case *kapi.ReplicationController:
+		changed := updatePodTemplate(t.Spec.Template, fn)
+		return reporter(changed), nil
+	case *kextensions.ReplicaSet:
+		changed := updatePodTemplate(&t.Spec.Template, fn)
+		return reporter(changed), nil
+	case *kextensions.DaemonSet:
+		changed := updatePodTemplate(&t.Spec.Template, fn)
+		return reporter(changed), nil
+	case *kextensions.Job:
+		changed := updatePodTemplate(&t.Spec.Template, fn)
+		return reporter(changed), nil
+	case *kextensions.Deployment:
+		changed := updatePodTemplate(&t.Spec.Template, fn)
+		return reporter(changed), nil
+	case *kapi.Pod:
+		changed := updatePodSpec(&t.Spec, fn)
+		return reporter(changed), nil
+	case *buildapi.BuildConfig:
+		var changed bool
+		if to := t.Spec.Output.To; to != nil && to.Kind == "DockerImage" {
+			changed = updateString(&to.Name, fn) || changed
+		}
+		for i, image := range t.Spec.Source.Images {
+			if image.From.Kind == "DockerImage" {
+				changed = updateString(&t.Spec.Source.Images[i].From.Name, fn) || changed
+			}
+		}
+		if c := t.Spec.Strategy.CustomStrategy; c != nil && c.From.Kind == "DockerImage" {
+			changed = updateString(&c.From.Name, fn) || changed
+		}
+		if c := t.Spec.Strategy.DockerStrategy; c != nil && c.From.Kind == "DockerImage" {
+			changed = updateString(&c.From.Name, fn) || changed
+		}
+		if c := t.Spec.Strategy.SourceStrategy; c != nil && c.From.Kind == "DockerImage" {
+			changed = updateString(&c.From.Name, fn) || changed
+		}
+		return reporter(changed), nil
+	}
+	// TODO: implement use of the generic PodTemplate accessor from the factory to handle
```

Comment:
Nope, just one.

On Wed, Jun 15, 2016 at 2:35 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/admin/migrate/images/images.go
> <https://github.com/openshift/origin/pull/9240#discussion_r67219247>:
>
> > +			if image.From.Kind == "DockerImage" {
> > +				changed = updateString(&t.Spec.Source.Images[i].From.Name, fn) || changed
> > +			}
> > +		}
> > +		if c := t.Spec.Strategy.CustomStrategy; c != nil && c.From.Kind == "DockerImage" {
> > +			changed = updateString(&c.From.Name, fn) || changed
> > +		}
> > +		if c := t.Spec.Strategy.DockerStrategy; c != nil && c.From.Kind == "DockerImage" {
> > +			changed = updateString(&c.From.Name, fn) || changed
> > +		}
> > +		if c := t.Spec.Strategy.SourceStrategy; c != nil && c.From.Kind == "DockerImage" {
> > +			changed = updateString(&c.From.Name, fn) || changed
> > +		}
> > +		return reporter(changed), nil
> > +	}
> > +	// TODO: implement use of the generic PodTemplate accessor from the factory to handle
>
> This is doing the mutation and is single resource so there can't be
> multiple hits returned.
>
> I haven't followed the final implementation, but I expected a PetSet to
> be allowed to contain multiple PodSpecs for various resources.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9240/files/326a588eab69d9160cff7758a91fd77c4f198898#r67219247>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p7TJd2_4HBRe7hKzYiGjktXFEjmuks5qMEXggaJpZM4IxnZA>
> .
>